src/rgw/rgw_rados.cc

```diff
@@ -8366,8 +8375,9 @@ int RGWRados::cls_bucket_list_ordered(RGWBucketInfo& bucket_info,
 
     // refresh the candidates map
     candidates.erase(candidates.begin());
-    if (! tracker.advance().at_end()) {
-      candidates[tracker.entry_name()] = tracker_idx;
+    tracker.advance();
+    if (tracker.next_candidate(candidates, tracker_idx)) {
```

Comment:
As a result, this is a conditional with side-effects.

Reply:
Or I should have said, it's an empty conditional where the effect is all due to side-effects.